core-ui/src/components/Predefined/Create/Certificates/Certificates.create.js

```diff
@@ -1,63 +1,189 @@
+            tooltipContent={t('certificates.tooltips.csr')}
+            propertyPath="$.spec.csr"
+            input={({ value, setValue }) => {
+              console.log('CSR', value);
+              return (
+                <FormTextarea
+                  compact
+                  required
+                  className="resize-vertical"
+                  onChange={e =>
+                    setValue(
+                      csrIsEncoded
+                        ? e.target.value
+                        : base64Encode(e.target.value),
+                    )
+                  }
+                  value={csrIsEncoded ? value : base64Decode(value || '')}
+                  placeholder={
+                    csrIsEncoded
+                      ? t('certificates.placeholders.encoded-csr')
+                      : t('certificates.placeholders.csr')
+                  }
+                />
+              );
+            }}
+          />
+        </>
+      )}
+      {!withCSR && (
+        <>
+          <ResourceForm.FormField
+            required
+            label={t('certificates.common-name')}
+            tooltipContent={t('certificates.tooltips.common-name')}
+            input={Inputs.Text}
+            propertyPath="$.spec.commonName"
+            inputProps={{
+              maxlength: 64,
+              placeholder: t('certificates.placeholders.common-name'),
+            }}
+          />
+          <ResourceForm.TextArrayInput
+            advanced
+            propertyPath="$.spec.dnsNames"
+            title={t('certificates.dns-names')}
+            inputProps={{
+              placeholder: t('certificates.placeholders.dns-names'),
+            }}
+          />
+        </>
+      )}
+      <IssuerRef
+        advanced
+        title={t('certificates.issuer')}
+        tooltipContent={t('certificates.tooltips.issuer')}
+        propertyPath="$.spec.issuerRef"
+      />
+      <ResourceForm.FormField
+        advanced
+        label={t('certificates.renew')}
+        tooltipContent={t('certificates.tooltips.renew')}
+        propertyPath="$.spec.renew"
+        input={Inputs.Switch}
+      />
+      <ResourceForm.FormField
+        advanced
+        label={t('certificates.existing-secret')}
+        tooltipContent={t('certificates.tooltips.existing-secret')}
+        input={() => (
+          <Switch
+            compact
+            onChange={e => setExistingSecret(!existingSecret)}
+            checked={existingSecret}
+          />
+        )}
+      />
+      {!existingSecret && (
+        <ResourceForm.FormField
+          advanced
+          label={t('certificates.secret-name')}
+          tooltipContent={t('certificates.tooltips.secret-name')}
+          propertyPath="$.spec.secretRef.name"
+          input={Inputs.Text}
+          inputProps={{
+            placeholder: t('certificates.placeholders.secret-name'),
+          }}
+        />
+      )}
+      {existingSecret && (
+        <SecretRef
```

Comment:
The search doesn't work as expected, I cannot see all namespaces there

Reply:
Also, shouldn't we only allow current namespace then?
<img width="380" alt="Screen Shot 2021-10-04 at 14 42 57" src="https://user-images.githubusercontent.com/40663133/135853378-421ec530-fe62-48ac-b0c6-2dc08c7b6392.png">

Reply:
i think this one might be related to specific implementation, not the api itself...